controllers/metal3.io/host_state_machine.go

```diff
@@ -227,6 +235,37 @@ func (hsm *hostStateMachine) checkInitiateDelete() bool {
 	return true
 }
 
+func (hsm *hostStateMachine) checkDetachedHost(info *reconcileInfo) (result actionResult) {
+	// If the detached annotation is set we remove any host from the
+	// provisioner and take no further action
+	// Note this doesn't change the current state, only the OperationalStatus
+	annotations := hsm.Host.GetAnnotations()
+	if annotations != nil {
+		if _, ok := annotations[metal3v1alpha1.DetachedAnnotation]; ok {
```

Comment:
nit: a `hasDetachedAnnotation()` convenience function wouldn't hurt.

Reply:
Done, I added a local function for now, but this seems to be a recurring pattern, so perhaps we can refactor to add a shared host.HasAnnotation() in future (I didn't attempt it here because it means moving some annotation constants which feels outside the scope of this series).